### PR TITLE
Generated Dockerfile optimization.

### PR DESCRIPTION
**Pull request by matejak**
The aim is to squash multiple statements into one, so the resulting image has smaller size and contains  fewer layers.

Statements to handle:
- [x] ADD
- [x] ENV
- [x] LABEL
- [x] RUN (the trickiest one)

### CODE REVIEW

File: generate-dockerfile.py [hunk 1]
```diff
@@ -72,31 +76,90 @@
...
+    envvar_value_lines = ['ENV'] + envvar_value_lines
+    env_statement = " \\\n{}".format(INDENTATION).join(envvar_value_lines)
+    return env_statement
+
+
+def _aggregate_by_destination(src_dest_pairs):
+    destinations = collections.defaultdict(set)
+    for src, dest in src_dest_pairs:
+        destinations[dest].add(src)
+    return destinations
```
Comment (jan-cerny) [inline generate-dockerfile.py]: This is really a great function! I love it.

**Comment by jan-cerny** [comment]
@openscap-ci add to whitelist

**Comment by mpreisler** [comment]
> * bzip2 package was removed as Python depends on it, so it is always present.

IMO if we used bzip2 directly we should depend on it directly. Python impl can change and then this whole thing breaks.

File: generate-dockerfile.py [hunk 2]
```diff
@@ -14,206 +20,374 @@
...
     ("install", "docker run --rm --privileged -v /:/host/ IMAGE sh /root/install.sh IMAGE"),
     ("run", "docker run -it --rm -v /:/host/ IMAGE sh /root/run.sh"),
-    ("help", "docker run --rm --privileged -v /usr/bin:/usr/bin -v /var/run:/var/run -v /lib:/lib -v /lib64:/lib64 -v /etc/sysconfig:/etc/sysconfig IMAGE sh /root/help.sh IMAGE")
-]
-packages = [
-    "bzip2",
-    "wget"
+    ("help", "docker run --rm --privileged -v /usr/bin:/usr/bin -v /var/run:/var/run -v /lib:/lib -v /lib64:/lib64 -v /etc/sysconfig:/etc/sysconfig IMAGE sh /root/help.sh IMAGE"),
 ]
+packages_for_cve_feed = {
```
Comment (matusmarhefka) [inline generate-dockerfile.py]: I tested both base images, Fedora and RHEL7, there is no `bzip2` package installed by default. I think there should be no problem and we can add `bzip2` package into the `packages_for_cve_feed`.

Comment (matejak) [reply generate-dockerfile.py]: When I try this on my machine, the baseimage already contains bzip2 at install-time, and when removing it, it wants to also remove `Python` and `openscap-daemon` (it is the same behaviour as on my Fedora 27, see below). So the conclusion is to keep `wget` and `bzip2` installed as they occupy little space and may be useful in the future.
```
$ sudo dnf remove bzip2                                                                                                                                                       
Dependencies resolved.
=======================
Package                                
=======================
Removing:
bzip2            ...
Removing dependent packages:
 flatpak-builder                       
...
 zstd                ...
Transaction Summary
======================
Remove  144 Packages
...
```

**Comment by jan-cerny** [comment]
This works OK for me.

**Pull request merged**